app/src/main/java/de/westnordost/streetcomplete/screens/main/bottom_sheet/MoveNodeFragment.kt

```diff
@@ -0,0 +1,200 @@
+    private val okButton get() = binding.okButton
+    private val okButtonContainer get() = binding.okButtonContainer
+
+    private val elementEditsController: ElementEditsController by inject()
+    private val questTypeRegistry: QuestTypeRegistry by inject()
+    private val overlayRegistry: OverlayRegistry by inject()
+    private val featureDictionaryFuture: FutureTask<FeatureDictionary> by inject(named("FeatureDictionaryFuture"))
+    private val mapDataWithEditsSource: MapDataWithEditsSource by inject()
+    private val countryBoundaries: FutureTask<CountryBoundaries> by inject(named("CountryBoundariesFuture"))
+    private val countryInfos: CountryInfos by inject()
+
+    override val elementKey: ElementKey by lazy { ElementKey(node.type, node.id) }
+
+    private lateinit var node: Node
+    private lateinit var editType: ElementEditType
+    private lateinit var displayUnit: MeasureDisplayUnit
+
+    private val hasChanges get() = getPosition() != node.position
+
+    interface Listener {
+        fun getMapPositionAt(screenPos: Point): LatLon?
+
+        fun onMovedNode(editType: ElementEditType, position: LatLon)
+    }
+    private val listener: Listener? get() = parentFragment as? Listener ?: activity as? Listener
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val args = requireArguments()
+        node = Json.decodeFromString(args.getString(ARG_NODE)!!)
+        editType = questTypeRegistry.getByName(args.getString(ARG_QUEST_TYPE)!!) as? OsmElementQuestType<*>
+            ?: overlayRegistry.getByName(args.getString(ARG_QUEST_TYPE)!!)!!
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        _binding = FragmentMoveNodeBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        okButton.setOnClickListener { onClickOk() }
+        binding.cancelButton.setOnClickListener { activity?.onBackPressed() }
+        val countryInfo = countryInfos.getByLocation(countryBoundaries.get(), node.position.longitude, node.position.latitude)
+        displayUnit = if (countryInfo.lengthUnits.firstOrNull() == LengthUnit.FOOT_AND_INCH)
+                MeasureDisplayUnitFeetInch(1)
+            else
+                MeasureDisplayUnitMeter(2)
+        binding.createNoteIconView.setImageResource(editType.icon)
+        highlightSimilarElements()
+    }
+
+    private fun highlightSimilarElements() {
+        val feature = featureDictionaryFuture.get().byTags(node.tags).isSuggestion(false).find().firstOrNull()
+        val tagsToFind = feature?.tags ?: node.tags
+        val mapData = mapDataWithEditsSource
+            .getMapDataWithGeometry(node.position.enclosingBoundingBox(MAX_MOVE_DISTANCE + 5.0))
```

Comment:
use `viewLifecycleScope.launch { ... }` with `withContext(Dispatchers.IO)`.

See also `MainFragment::showHighlightedElements`, try to mimic this behavior.

Actually, better yet, ... while this solution with using the feature dictionary is clever, it is inconsistent to the highlighting done while the quest is displayed. E.g. the "bench backrest" quest shows other benches **and** picnic tables. Now if a user taps "at slightly different position", only other benches are shown. Its even much worse for any shop-related quests, because the `MoveNodeFragment` would only show exactly the same shop type as the user is moving. This **will** 100% lead to confusion.

I believe it would be possible with only little refactoring to just let `MainFragment` show the highlighted elements of in whatever quest's context this fragment was started with. After all, the quest type is passed to this fragment. All `MainFragment::showHighlightedElements` needs is a position and a `getHighlightedElements` of a quest type currently.

Reply:
Well, then the simplest solution would be to use `showInBottomSheet(MoveNodeFragment.create(editType, node), clearPreviousHighlighting = false)`, and hide overlay only if `editType !is Overlay`.
Only (potential) issue here is that the quest pin remains at the original position,  though indicating starting position is not necessarily bad.

Reply:
Hm, I didn't think of that. This might be the easiest solution. I can't think of any downside.

And original quest pin could just be hidden manually in the function that shows the MoveNodeFragment, no?

Reply:
Adjusted, so this is now happening.